docs/manual/src/udl/enumerations.md

```diff
@@ -17,4 +18,25 @@ enum Animal {
 };
 ```
 
-Note that enumerations with associated data are not yet supported.
+Enumerations with associated data require a different syntax,
+due to the limitations of using WebIDL as the basis for UniFFI's interface language.
+An enum like this in Rust:
+
+```rust
+enum IpAddr {
+  V4 {q1: u8, q2: u8, q3: u8, q4: u8},
+  V6 {addr: string},
+}
+```
+
+Can be exposed in the UDL file with:
+
+```idl
+[TaggedUnion]
```

Comment:
Please feel welcome to propose your favourite colour for this bikeshed. I chose "tagged union" because that's the canonical name of the Wikipedia page about this concept.

Reply:
heh - see above - I mildly prefer "tagged enum" given we are in rust.